pkg/kademlia/dht.go

```diff
@@ -5,37 +5,178 @@ package kademlia
 
 import (
 	"context"
+	"crypto/rand"
+	"fmt"
+	"strings"
 
-	"storj.io/storj/protos/overlay"
+	bkad "github.com/coyle/kademlia"
+	"github.com/zeebo/errs"
+
+	proto "storj.io/storj/protos/overlay"
 )
 
+// NodeErr is the class for all errros petaining to node operations
```

Comment:
"errros" not that it really matters though lol

Reply:
it matters!